### ISSUE TITLE
Query Execute Callback hangs - unexpected 352583681

### ISSUE THREAD

**Issue by BradenBagby** [hgourvest/node-firebird#209]
I can successfully connect to Firebird Database. When I perform a query or execute, the callback is only called when the SQL statement is wrong.

For example:
SELECT * FROM table - callback never called
SELECT * FROM fake_made_up_table - callback is called with correct error message

I have traced this bug to node-firebird's index.js > socket's onData callback > decodeResponse > parseOpResponse > and the switch statement on op.
 
when it errors, op is '352583681' resulting in the default, which never calls the callback. Node-firebird seems to be sending and receiving some data, but in the middle of any correct query, it will throw 'Unexpected 352583681', which makes me wonder if it is parsing the data wrong?

It will throw this error multiple times for a single query, usually between 2-5 times.

I know my Firebird Databse setup is correct as I have successfully connected and queried it in 2 other clients.

Here is my connection code with some stuff removed just to be thorough 

`
  var options = {};

  options.host = '*****'; //removed 
  options.port = 3050;
  options.database = '***/ejdb.fdb'; //removed
  options.user = '******';
  options.password = '*****';
  options.lowercase_keys = false; // set to true to lowercase keys
  options.role = null;            // default
  options.pageSize = 4096;        // default when creating database

  Firebird.attach(options, function (err, db) {
      if (err) {
          console.log(err);
          return;
      }

      console.log("successful connection");

      db.query("SELECT * from TABLE", function (err, result) {
        console.log("result reached")
        console.log(result)
        db.detach();
      });

  });
`

Here is the switch statement from node-firebird that throws the 'Unexpected 352583681'
`	op = data.readInt();

		switch (op){
			case isc_arg_end:
				return response;
			case isc_arg_gds:
				num = data.readInt();
				if (!num)
					break;

				item = { gdscode: num };

				if (response.status)
					response.status.push(item);
				else
					response.status = [item];

				break;
			case isc_arg_string:
			case isc_arg_interpreted:
			case isc_arg_sql_state:
				if (item.params) {
					var str = data.readString(DEFAULT_ENCODING);
					item.params.push(str);
				} else
					item.params = [data.readString(DEFAULT_ENCODING)];

				break;
			case isc_arg_number:
				num = data.readInt();

				if (item.params)
					item.params.push(num);
				else
					item.params = [num];

				if (item.gdscode === isc_sqlerr)
					response.sqlcode = num;

				break;
			default:
				throw new Error('Unexpected: ' + op);`

**Comment by mreis1** [comment]
Facing the same issue. As a temporary workaround I had to downgrade from 0.9.3 to 0.8.10.

**Comment by Jaime02** [comment]
I am still suffering this bug in the version (v1.1.9) 🙁 Which info should I add to the bug report?
